### PR TITLE
Feature/assignment 6

### PR DESCRIPTION
**Pull request by nikhil1-successive**

### CODE REVIEW

File: src/assignment-6/Ques1/Ques1.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// 1.Create a React application that has multiple pages represented as components. Implement lazy loading for these components using React's lazy and Suspense features. When a user navigates to a particular page, the component should be loaded lazily, improving the application's initial loading time.
+import React from "react";
+import { Suspense, lazy } from "react";
+
+const Home = lazy(() => import('./Home'))
+const About = lazy(() => import('./About'))
+
+function Ques1() {
```
Comment (shahbaz-successive) [inline src/assignment-6/Ques1/Ques1.jsx:8]: User should navigate to page when clicked home or about section. So try to implement a navigation to separate page.

File: src/assignment-6/Ques2/Ques2.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        console.error(error);
+    };
+
+    return error ? <p>Failed To Load Component. Try Again.</p> : children;
+};
+
+const LazyComponent1 = React.lazy(() => import('../Ques1/Home.jsx'));
+const LazyComponent2 = React.lazy(() => import('../Ques1/About.jsx'));
+
+const Ques2 = () => (
```
Comment (shahbaz-successive) [inline src/assignment-6/Ques2/Ques2.jsx:19]: Once question 1 navigation is resolved update this too accordingly

File: src/assignment-6/Ques2/Ques2.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// 2.Extend Question 1 to include error handling. If a lazy-loaded component fails to load, display a user-friendly error message to the user instead of crashing the application. Implement this using the ErrorBoundary component.
+
+import React, { useState } from 'react';
+
+const ErrorBoundary = ({ children }) => {
+    const [error, setError] = useState(false);
+
+    const componentDidCatch = (error) => {
```
Comment (shahbaz-successive) [inline src/assignment-6/Ques2/Ques2.jsx:8]: This is unused code

File: src/assignment-6/Ques3/Ques3.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import Ques3 from './Ques3';
+
+test('increment btn click so count = count + 1', () => {
```
Comment (shahbaz-successive) [inline src/assignment-6/Ques3/Ques3.test.js:5]: Test cases not running on my system, is it working on yours?

File: src/assignment-6/Ques4/Ques4.test.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import Ques4 from './Ques4';
+
+describe('Ques4', () => {
```
Comment (shahbaz-successive) [inline src/assignment-6/Ques4/Ques4.test.js:5]: Check and run all test cases as it's not working on my system. Check for Ques5 and 6 as well.

**Review by shahbaz-successive — CHANGES_REQUESTED** [review]
Feedback changes

File: src/assignment-6/Ques3/Ques3.test.js (hunk 4 above)
Comment (nikhil1-successive) [reply src/assignment-6/Ques3/Ques3.test.js:5]: Yes it is working.

File: src/assignment-6/Ques1/Ques1.jsx (hunk 1 above)
Comment (nikhil1-successive) [reply src/assignment-6/Ques1/Ques1.jsx:8]: done

File: src/assignment-6/Ques2/Ques2.jsx (hunk 2 above)
Comment (nikhil1-successive) [reply src/assignment-6/Ques2/Ques2.jsx:19]: done

File: src/assignment-6/Ques2/Ques2.jsx (hunk 3 above)
Comment (nikhil1-successive) [reply src/assignment-6/Ques2/Ques2.jsx:8]: done

File: src/assignment-6/Ques4/Ques4.test.js (hunk 5 above)
Comment (nikhil1-successive) [reply src/assignment-6/Ques4/Ques4.test.js:5]: done

**Review by shahbaz-successive — APPROVED** [review]
👍

**Pull request merged**